### DISCUSSION TITLE
Can I convert seurat v5 to v4 object?

### DISCUSSION THREAD

**Discussion by Yijia-Jiang** [satijalab/seurat#8350] [General]
Hello, thank you for the tool. I am using seuratv5 on server, but find many packages are unable to run for seuratv5 object. I want to convert into seurat v4 and run packages on my local laptop. Is there an easy way to do so? 

I got the following error when reading seurat v5 object:
Error in UseMethod(generic = "VariableFeatures", object = object) : 
  no applicable method for 'VariableFeatures' applied to an object of class "Assay5"
Error during wrapup: cannot get a slot ("slots") from an object of type "NULL"
Error: no more error handlers available (recursive errors?); invoking 'abort' restart

**Comment by yuanG2000** [comment]
Hi， I ran into that same problem. did you find out the solution? thx a lot
I save my data in seuratobject V5,but now I install seurat V4, I cannot read in my data now.

**Reply by samuel-marsh** [reply]
See here:
https://samuel-marsh.github.io/scCustomize/articles/Object_Conversion.html#convert-seurat-assay-type-within-an-object

**Reply by yuanG2000** [reply]
thank you! i have a try

**Reply by cystone** [reply]
I got the following error when using scCustomize::Convert_Assay()
sce <- Convert_Assay(seurat_object = sce, convert_to = "V3")
Converting assay "RNA" from Assay5 to Assay.
Error in .subset2(x, i, exact = exact) : subscript out of bounds

Here is my scripts:
options(Seurat.object.assay.version = "v3")
 sce[["RNA"]] = as(object = sce[["RNA"]], Class = "Assay5")
  HGNC.updated = HGNChelper::checkGeneSymbols(rownames(sce))
  HGNC.updated = dplyr::mutate(HGNC.updated, symbol = ifelse(Approved, Suggested.Symbol, x))
  sce = RenameGenesSeurat(sce,newnames = HGNC.updated$symbol)
  sce <- Convert_Assay(seurat_object = sce, convert_to = "V3")

**Reply by samuel-marsh** [reply]
Hi @cystone,

please post issue to scCustomize repo and happy to help!

Best,
Sam
